### PR TITLE
(Fixes #224) Spark*Activity should allow setting parameters for spark jobs

### LINKED ISSUE #224: Spark*Activity should allow setting parameters for spark jobs
Right now, we pass in `-x` to `run-spark-step.sh` to allocate full resources.  We should allow specifying executor memory and/or cores as well as generic spark configuration options.

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

**Comment by realstraw** [comment]
So this still does not allow us to specify the additional spark options. We need to update `com.krux.hyperion.activity.SparkStep` as well to allow parsing these additional options in.

File: core/src/main/scala/com/krux/hyperion/activity/SparkJobActivity.scala [hunk 1]
```diff
@@ -63,14 +67,45 @@ case class SparkJobActivity private (
...
+    sparkOptions: Seq[String] = sparkOptions,
+    sparkConfig: Map[String, String] = sparkConfig
+  ): SparkJobActivity = new SparkJobActivity(id, scriptRunner, jobRunner, jarUri, mainClass, args,
+    hadoopQueue, preActivityTaskConfig, postActivityTaskConfig, inputs, outputs, runsOn, dependsOn,
+    preconditions, onFailAlarms, onSuccessAlarms, onLateActionAlarms, attemptTimeout, lateAfterTimeout,
+    maximumRetries, retryDelay, failureAndRerunMode, sparkOptions, sparkConfig)
+
   def objects: Iterable[PipelineObject] = inputs ++ outputs ++ runsOn.toSeq ++ dependsOn ++ preconditions ++ onFailAlarms ++ onSuccessAlarms ++ onLateActionAlarms ++ preActivityTaskConfig.toSeq ++ postActivityTaskConfig.toSeq
 
+  private def sparkSettings: Seq[String] = sparkOptions ++ sparkConfig.flatMap { case (k, v) => Seq("--conf", k, v) }
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/activity/SparkJobActivity.scala]: it should be `s"$k=$v"`.

File: core/src/main/scala/com/krux/hyperion/activity/SparkStep.scala [hunk 2]
```diff
@@ -5,20 +5,31 @@ import com.krux.hyperion.common.S3Uri
...
   def withArguments(arg: String*) = this.copy(args = args ++ arg)
-
-  override def toString: String = (scriptRunner.toSeq ++ jobRunner.toSeq ++ Seq(jarUri.toString) ++ mainClass.map(_.toString).toSeq ++ args).mkString(",")
+  def withSparkOption(option: String*) = this.copy(sparkOptions = sparkOptions ++ option)
+  def withSparkConfig(key: String, value: String) = this.copy(sparkConfig = sparkConfig + (key -> value))
+
+  override def toString: String = Seq(
+    Seq(scriptRunner, jobRunner),
+    sparkOptions,
+    sparkConfig.flatMap { case (k, v) => Seq("--conf", k, v) }.toSeq,
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/activity/SparkStep.scala]: it should be `s"$k=$v"`.

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/activity/SparkStep.scala]: Duh.  Good catch.  Fixed.

File: core/src/main/scala/com/krux/hyperion/activity/SparkJobActivity.scala (hunk 1 above)
Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/activity/SparkJobActivity.scala]: Duh.  Good catch.  Fixed.

**Comment by realstraw** [comment]
:+1: :sparkles: :sparkles: :sparkles:

**Pull request closed without merging**